lib/connections/util.go

```diff
@@ -71,12 +71,10 @@ func getHostPortsForAllAdapters(port int) []string {
 	portStr := strconv.Itoa(port)
 
 	for _, network := range nets {
-		// Only IPv4 addresses, as v6 link local require an interface identifiers to work correctly
-		// And non link local in theory are globally routable anyway.
-		if network.IP.To4() == nil {
-			continue
-		}
-		if network.IP.IsLinkLocalUnicast() || (isV4Local(network.IP) && network.IP.IsGlobalUnicast()) {
+		// Only accept IPv4 link-local unicast and the private ranges defined in RFC 1918 and RFC 4193
+		// IPv6 link-local addresses require an interface identifier to work correctly
+		if (network.IP.To4() != nil && network.IP.IsLinkLocalUnicast()) ||
+			(network.IP.IsPrivate() && network.IP.IsGlobalUnicast()) {
```

Comment:
I think the `IsGlobalUnicast` is redundant here, may have been so before as well

Reply:
I'm not sure if it's possible to misconfigure an interface so that we end up with a broadcast/multicast/whatever here. But I think it's safe to skip this check.